### PR TITLE
Multiple major changes

### PR DESCRIPTION
**Pull request by bubby932**
- Moved all commands into separate files.
- Moved processing of triggers and insults into separate files.
- Moved Mention message into separate file.
- Added a (wip) cooldown to sp!dm.
- Minor change to README.md due to security scanning guidelines.
- Fixed missing semicolons.
- Updated command handling to use a switch statement instead of an if/else chain.
- Added a configuration value to change how often the status updates, default 15.
- Updated code comments and added spaces where necessary.
- Added line breaks between multiple code blocks for readability.
- Added #region blocks for ease of reading.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,284 +48,117 @@ process
...
-      .setColor('#04d384')
-      .setTitle(langstitle)
-      .setAuthor(langsauthor)
-      .setDescription(langlist.toString())
-      .setURL('https://spbot.ml/')
-      .addField(langsfield1heading, langsfield1)
-      .setImage('https://www.spbot.ml/suicideicon.png')
-    message.channel.send({ embeds: [langs] });
-  }
+  switch(command) {
```
Comment (CatusKing) [inline index.js:114]: bad ise a for loop on a n object of commands

File: index.js [hunk 2]
```diff
@@ -55,284 +48,117 @@ process
...
-        .setDescription(`Roundtrip latency is ${Math.floor(msg.createdTimestamp - message.createdTimestamp)}ms \nAPI Latency is ${Math.round(client.ws.ping)}ms`)
-      msg.edit({ embeds: [ping] });
-    });
-    return;
-  }
+  // #region Commands
+  // Ping command
+  
+  if (LCM === '(ping' || LCM === ')ping' || LCM === '\\ping' || LCM === '~ping' || LCM === '|ping' || LCM === '!ping' || LCM === '?ping' || LCM === '.ping' || LCM === '$ping' || LCM === '%ping' || LCM === '-ping' || LCM === '--ping' || LCM === '=ping' || LCM === '+ping' || LCM === '_ping' || LCM === '/ping' || LCM === '&ping' || LCM === '--ping' || LCM === "`ping" || LCM === 'sp!ping')
+    return require('./commands/ping')(message, lang);
```
Comment (CatusKing) [inline index.js:107]: stop reading the files upon every call upon the command files
use a basic command handler

**Review by CatusKing — CHANGES_REQUESTED** [review]

**Comment by bubby932** [comment]
@CatusKing I'll resolve those tomorrow, however for the switch statement i was intentionally using it to cut down on the need for loops, are you sure you want me to change that?

**Comment by bubby932** [comment]
also the files are cached on the first read, they are not re read every time require() is called. 
image

**Comment by CatusKing** [comment]
> also the files are cached on the first read, they are not re read every time require() is called. image

hmmm interesting didn't know but for me i've been taught its better practice to use an object with a simple for loop or the Collection offers a .get(key)

**Comment by bubby932** [comment]
> > also the files are cached on the first read, they are not re read every time require() is called. image
> 
> hmmm interesting didn't know but for me i've been taught its better practice to use an object with a simple for loop or the Collection offers a .get(key)

alright

**Comment by Killerjet101** [comment]
Those are older reviews as they've all been fixed in the canary branch

**Comment by CPlusPatch** [comment]
So we can close this, since it's already been merged in canary

**Pull request closed without merging**